src/lib/caching/stores/ChannelPinsStore.ts

```diff
@@ -0,0 +1,81 @@
+import { RequestOptions, Routes } from '@klasa/rest';
+import { ProxyCache } from '@klasa/cache';
+
+import type { APIMessageData } from '@klasa/dapi-types';
+import type { Client } from '../../client/Client';
+import type { Message } from '../structures/messages/Message';
+import type { GuildTextChannel } from '../structures/channels/GuildTextChannel';
+import type { DMChannel } from '../structures/channels/DMChannel';
+
+/**
+ * The store for the pins the channel has.
+ * @since 0.0.4
+ */
+export class ChannelPinsStore extends ProxyCache<string, Message> {
+
+	/**
+	 * The {@link Client client} this store belongs to.
+	 * @since 0.0.4
+	 */
+	public readonly client: Client;
+
+	/**
+	 * The {@link GuildTextChannel guild channel} or {@link DMChannel DM channel} this store belongs to.
+	 * @since 0.0.4
+	 */
+	public readonly channel: GuildTextChannel | DMChannel;
+
+	/**
+	 * Builds the store.
+	 * @since 0.0.4
+	 * @param channel The {@link GuildTextChannel guild channel} or {@link DMChannel DM channel} this store belongs to.
+	 */
+	public constructor(channel: GuildTextChannel | DMChannel, keys: string[]) {
+		super(channel.messages, keys);
+		this.client = channel.client;
+		this.channel = channel;
+	}
+
+	/**
+	 * Pins a message to the channel.
+	 * @since 0.0.4
+	 * @param id The {@link Message#id message id} you want to pin
+	 * @param requestOptions The additional request options.
+	 * @see https://discord.com/developers/docs/resources/channel#add-pinned-channel-message
+	 */
+	public async add(id: string, requestOptions: RequestOptions = {}): Promise<Message> {
+		const entry = await this.client.api.put(Routes.channelPin(this.channel.id, id), { ...requestOptions }) as APIMessageData;
+		this.set(entry.id);
+
+		// eslint-disable-next-line dot-notation
+		return this.channel.messages['_add'](entry);
```

Comment:
Reading the docs would be a good idea, this entire block is wrong
![image](https://user-images.githubusercontent.com/17960496/85622231-3d0b5780-b66f-11ea-89ae-625bcddddbe5.png)